src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.cs

```diff
@@ -79,7 +79,8 @@ private PointsToAnalysis(PointsToAnalysisDomain analysisDomain, PointsToDataFlow
         }
 
         internal static bool ShouldBeTracked(ITypeSymbol typeSymbol) => typeSymbol.IsReferenceTypeOrNullableValueType() ||
-            typeSymbol is ITypeParameterSymbol typeParameter && !typeParameter.IsValueType;
+            typeSymbol?.HasValueCopySemantics() == false ||
```

Comment:
The change here feels a bit conflicting for `string` types as we do want to track them, but `HasValueCopySemantics` returns false for it. Can we instead just have a new extension method on `ITypeSymbol` that explicitly checks for ref like types, say `IsRefLikeType()`, which delegates to the property when available, and returns false otherwise?

Reply:
@mavasani perhaps I did not understand the purpose of `HasValueCopySemantics`. From my understanding, it is incorrect to say, that `System.String` type has value copy semantics. Rather it is an immutable reference type.

Reply:
The helper method just intends to check if assigning a variable of the given type, say `a = b` causes `a` and `b` to  become reference equals or value equals.

Reply:
@mavasani seems quite unfortunate in terms of naming and the expectations. I'll probably just introduce `IsPassedByReference` and use that then.

Reply:
Reverted this change in favor of explicitly checking for `IsRefLikeType ` in `PointsToAnalysis` and `DisposeDataFlowOperationVisitor`